staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor.go

```diff
@@ -141,12 +143,75 @@ func (s *serializer) unmarshal(data []byte, into interface{}) (strict, lax error
 	if u, ok := into.(runtime.Unstructured); ok {
 		var content map[string]interface{}
 		defer func() {
-			// TODO: The UnstructuredList implementation of SetUnstructuredContent is
-			// not identical to what unstructuredJSONScheme does: (1) it retains the
-			// "items" key in its Object field, and (2) it does not infer a singular
-			// Kind from the list's Kind and populate omitted apiVersion/kind for all
-			// entries in Items.
-			u.SetUnstructuredContent(content)
+			switch u := u.(type) {
+			case *unstructured.UnstructuredList:
+				// UnstructuredList's implementation of SetUnstructuredContent
+				// produces different objects than those produced by a decode using
+				// UnstructuredJSONScheme:
+				//
+				//   1. SetUnstructuredContent retains the "items" key in the list's
+				//      Object field. It is omitted from Object when decoding with
+				//      UnstructuredJSONScheme.
+				//   2. SetUnstructuredContent does not populate "apiVersion" and
+				//      "kind" on each entry of its Items
+				//      field. UnstructuredJSONScheme does, inferring the singular
```

Comment:
Is this behavior practical to keep. it's very convenient for consumers and I worry about breaking it for actors like controller-runtime.

Reply:
Oh, I see. This is you making them match. nevermind.